tools/swift-api-digester/swift-api-digester.cpp

```diff
@@ -1106,6 +1100,12 @@ case SDKNodeKind::X:                                                           \
 // Recursively construct a node that represents a type, for instance,
 // representing the return value type of a function decl.
 static SDKNode *constructTypeNode(SDKContext &Ctx, Type T) {
+  // For generic function type, build from its canonical version.
+  if (auto GFT = dyn_cast<GenericFunctionType>(T.getPointer())) {
+    if (!GFT->isCanonical())
```

Comment:
Sorry, what I meant is that I think constructTypeNode() should always be called with a CanType. The caller should canonicalize the type before passing it in.